pkg/steps/pod.go

```diff
@@ -193,42 +193,48 @@ func PodStep(name string, config PodStepConfiguration, resources api.ResourceCon
 	}
 }
 
-func (s *podStep) generatePodForStep(image string, containerResources coreapi.ResourceRequirements) (*coreapi.Pod, error) {
-	envMap, err := downwardapi.EnvForSpec(s.jobSpec.JobSpec)
+func generateBasePod(jobSpec *api.JobSpec, containerResources coreapi.ResourceRequirements) (*coreapi.Pod, error) {
+	envMap, err := downwardapi.EnvForSpec(jobSpec.JobSpec)
 	if err != nil {
 		return nil, err
 	}
-	pod := &coreapi.Pod{
+	return &coreapi.Pod{
 		ObjectMeta: meta.ObjectMeta{
-			Name: s.config.As,
 			Labels: trimLabels(map[string]string{
 				PersistsLabel:    "false",
-				JobLabel:         s.jobSpec.Job,
-				BuildIdLabel:     s.jobSpec.BuildID,
-				ProwJobIdLabel:   s.jobSpec.ProwJobID,
+				JobLabel:         jobSpec.Job,
+				BuildIdLabel:     jobSpec.BuildID,
+				ProwJobIdLabel:   jobSpec.ProwJobID,
 				CreatedByCILabel: "true",
 			}),
 			Annotations: map[string]string{
-				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
-				annotationContainersForSubTestResults: s.name,
+				JobSpecAnnotation: jobSpec.RawSpec(),
 			},
 		},
 		Spec: coreapi.PodSpec{
-			ServiceAccountName: s.config.ServiceAccountName,
-			RestartPolicy:      coreapi.RestartPolicyNever,
+			RestartPolicy: coreapi.RestartPolicyNever,
 			Containers: []coreapi.Container{
 				{
-					Image:                    image,
 					Env:                      decorate.KubeEnv(envMap),
-					Name:                     s.name,
-					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
 					Resources:                containerResources,
 					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
 				},
 			},
 		},
-	}
+	}, nil
+}
 
+func (s *podStep) generatePodForStep(image string, containerResources coreapi.ResourceRequirements) (*coreapi.Pod, error) {
+	pod, err := generateBasePod(s.jobSpec, containerResources)
+	if err != nil {
+		return nil, err
+	}
+	pod.ObjectMeta.Name = s.config.As
```

Comment:
All of these look like the same post-generation edits as in the other file. Why not pass them in or otherwise DRY them out?